pkg/credentialprovider/apis/criauth/types.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package criauth
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CRIAuthPluginConfig is the configuration containing information about
+// each CRI auth plugin. Kubelet reads this configuration from disk and enables
+// each plugin as specified by the CRIAuthPlugin type.
+type CRIAuthPluginConfig struct {
+	metav1.TypeMeta
+
+	// Plugins is a list of CRI auth plugins that will be enabled by the kubelet
```

Comment:
I wish there is a linter for API fields. For consistence I'd suggest to do lower case here. Sorry for so many nit comments

```suggestion
	// plugins is a list of CRI auth plugins that will be enabled by the kubelet
```

Reply:
> Sorry for so many nit comments

Not a problem, they're needed :)